policy-service/src/main/java/uk/gov/gchq/palisade/service/policy/PolicyApplication.java

```diff
@@ -16,26 +16,133 @@
 public class PolicyApplication {
     private static final Logger LOGGER = LoggerFactory.getLogger(PolicyApplication.class);
 
+    private final Set<RunnableGraph<?>> runners;
+    private final Materializer materializer;
+    private final Executor executor;
+    private final PolicyServiceCachingProxy service;
+    private final PolicyConfiguration policyConfig;
+    private final UserConfiguration userConfig;
+    private final ResourceConfiguration resourceConfig;
+
+    /**
+     * Autowire Akka objects in constructor for application ready event
+     *
+     * @param runners        collection of all Akka {@link RunnableGraph}s discovered for the application
+     * @param materializer   the Akka {@link Materializer} configured to be used
+     * @param service        specifically policyServiceCachingProxy used for pre-population
+     * @param policyConfig   resourceConfig used to create the policy object used in pre-population
+     * @param userConfig     resourceConfig used to create the user object used in pre-population
+     * @param resourceConfig resourceConfig used to create the resource object used in pre-population
+     * @param executor       an executor for any {@link CompletableFuture}s (preferably the application task executor)
+     */
+    public PolicyApplication(
+            final Collection<RunnableGraph<?>> runners,
+            final Materializer materializer,
+            final PolicyServiceCachingProxy service,
+            final PolicyConfiguration policyConfig,
+            final UserConfiguration userConfig,
+            final ResourceConfiguration resourceConfig,
+            @Qualifier("threadPoolTaskExecutor") final Executor executor) {
+        this.runners = new HashSet<>(runners);
+        this.materializer = materializer;
+        this.service = service;
+        this.policyConfig = policyConfig;
+        this.userConfig = userConfig;
+        this.resourceConfig = resourceConfig;
+        this.executor = executor;
+    }
+
+    /**
+     * Application entrypoint, creates and runs a spring application, passing in the given command-line args
+     *
+     * @param args command-line arguments passed to the application
+     */
     public static void main(final String[] args) {
-        LOGGER.debug("PolicyApplication started with: {} {} {}", PolicyApplication.class.toString(), "main", Arrays.toString(args));
+        LOGGER.debug("PolicyApplication started with: {}", (Object) args);
         new SpringApplicationBuilder(PolicyApplication.class).web(WebApplicationType.SERVLET)
                 .run(args);
     }
 
+    /**
+     * Init post construct.
+     * Pre-population profile used in the KafkaContractTests for pre-populating the cache.
+     * Example profile used for pre-populating the cache for the example
+     */
+    @EventListener(ApplicationReadyEvent.class)
+    @Profile({"pre-population, example"})
```

Comment:
I don't like that this is referencing test stuff and example stuff... Can it not just run for all profiles? What if i create a prepop profile with a different name?

Reply:
you can, but why would you always want the prepopulation to run? 
If you run the service for the first time you would wonder why a bunch of users have been added

Reply:
I'd prefer a boolean prepop/no-prepop config value to switch the behavior on and off if we wanted to not run prepop.
Otherwise if we have a dozen different prepops, this has to be updated for each one.